File: python/374_Additive_persistence.py
def persistence(number):
    sums = 0
    while number > 9:
        number = sum_digits(number)
        sums += 1
    return sums

def sum_digits(n):
    total = 0
    while n:
        total += n % 10
        n //= 10
    return total

print(persistence(1234))

File: rust/src/374_additive_persistence.rs
fn sum_digits(n: i32) -> i32 {
    let mut total = 0;
    let mut num: i32 = n;
    while num != 0 {
        total += num % 10;
        num /= 10;
    }
    total
}

fn persistence(number: i32) -> i32 {
    let mut num: i32 = number;
    let mut sums = 0;
    while num > 9 {
        num = sum_digits(num);
        sums += 1;
    }
    sums
}

fn main() {
    let a : i32 = 1234;
    let result = persistence(a);
    println!("{}", result);
}